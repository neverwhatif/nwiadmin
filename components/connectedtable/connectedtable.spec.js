import React from 'react';
import { shallow } from 'enzyme';

import { ConnectedTableComponent } from './index';

jest.mock('nwiadmin/components/connectedlist', () => ({ ConnectedListComponent: 'ConnectedListComponent' }));
jest.mock('nwiadmin/components/table', () => 'Table');

const mockLocation = {
    pathname: '/lorem',
    search: '',
};

describe('ConnectedTable', () => {
    const component = shallow(<ConnectedTableComponent remote="lorem" location={mockLocation} />);

    it('should render', () => {
        expect(component).toMatchSnapshot();
    });
    it('should use render prop', () => {
        const listComponent = component.find('ConnectedListComponent').first();
        const deepList = listComponent.prop('renderList')({ lorem: 1 });

        expect(deepList.type).toBe('Table');
        expect(deepList.props.lorem).toBe(1);
    });
});
