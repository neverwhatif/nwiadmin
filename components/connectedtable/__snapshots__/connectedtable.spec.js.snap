// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedTable should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConnectedTableComponent
    location={
      Object {
        "pathname": "/lorem",
        "search": "",
      }
    }
    remote="lorem"
    transformer={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "location": Object {
        "pathname": "/lorem",
        "search": "",
      },
      "remote": "lorem",
      "renderList": [Function],
      "transformer": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": "ConnectedListComponent",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "location": Object {
          "pathname": "/lorem",
          "search": "",
        },
        "remote": "lorem",
        "renderList": [Function],
        "transformer": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": "ConnectedListComponent",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
